### ISSUE TITLE
Enhancement: Negate lua altogether

### ISSUE THREAD

**Issue by isakube** [KadeArchive/Kade-Engine#1581]
## What is your suggestion, and why should it be implemented?
Lua is stupid. In particular, linc_luajit. So instead we just add song events for all the modchart functions.

Song Events
- Work on all platforms
- Are easy to implement

Lua Modcharts
- Only work on windows
- Are in a completely different language
- Stupid
- No Brackets (seriously wtf why is that)

already working on this, too. just putting this here in case anybody wants to race. :)

**Comment by Jaduplex** [comment]
No. Modcharts and song events can coexist. Song events aren’t advanced enough to replace modcharts, and modcharts are too advanced to replace song events.

**Comment by isakube** [comment]
> Song events aren’t advanced enough to replace modcharts

I can understand this, since all of the modchart functions require more than one input, and uhh
![lack of multiple inputs](https://user-images.githubusercontent.com/68293280/128207949-7d6b53d0-6357-4af9-a31c-7c51e8071593.png)
Perhaps we could add one or two, like Asset Changes (`changeDadCharacter()`, `changeBoyfriendCharacter()`), which are really simple and don't require several inputs.
ALTHOUGH
We could parse arguments with commas or spaces

> modcharts are too advanced to replace song events.

Still trying to wrap my head around what this means because today's a smooth brain day for me

I think you're saying that the method of recreating a song event through a modchart is more complex than song events currently are. idk. im dumb.

But yeah i can see what you mean. 🅰️

**Comment by Jaduplex** [comment]
yeah, asset changes are pretty easy, ill see if i can funny pr it

**Comment by Futuremappermydud** [comment]
I’d be fine with hscript AND lua mod charts. Based on the file extension it would switch, can use existing code and provide flexibility for those who prefer haxe over lua.

**Comment by isakube** [comment]
> I’d be fine with hscript AND lua mod charts. Based on the file extension it would switch, can use existing code and provide flexibility for those who prefer haxe over lua.

.lua for lua
.hsc / .hscript for hscript

**Comment by Futuremappermydud** [comment]
> > I’d be fine with hscript AND lua mod charts. Based on the file extension it would switch, can use existing code and provide flexibility for those who prefer haxe over lua.
> 
> .lua for lua
> .hsc / .hscript for hscript

Yes

**Comment by ThatOneCalculator** [comment]
This, PLEASE! It sucks not having modcharts on Linux

**Comment by Futuremappermydud** [comment]
I’m working on lua and hscript support for my engine might pr into kade
